lib/ansible/modules/cloud/misc/virt.py

```diff
@@ -450,6 +461,11 @@ def core(module):
             module.fail_json(msg = "state change requires a guest specified")
 
         res['changed'] = False
```

Comment:
The whole block need to moved outside the `if state:` condition, otherwise autostart parameter won't be updated when `state` option isn't used. In case `autostart` couldn't be used without `autostart`, add a `required_if` parameter to `AnsibleModule`.